src/components/AddMovie.jsx

```diff
@@ -1 +1,72 @@
-// implement AddMovie component here
+import React, { Component } from 'react';
+
+class AddMovie extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      subtitle: '',
+      title: '',
+      imagePath: '',
+      storyline: '',
+      rating: 0,
+      genre: 'action',
+    };
+
+    this.handle = this.handle.bind(this);
+    this.includeMovie = this.includeMovie.bind(this);
+  }
+
+  resetState() {
+    this.setState(
+      {
+        subtitle: '',
+        title: '',
+        imagePath: '',
+        storyline: '',
+        rating: 0,
+        genre: 'action',
+      },
+    );
+  }
+
+  handle(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
```

Comment:
Tá tendo um probleminha aqui, quando tá vindo a informação do _event_, o _name_ não tá chegando como esperado, pra solucionar isso quando for chamar a função você passa o _name_ da chave do _state_, e usa ela aqui. E já acrescendo que vai dar conflito no _rating_, pois ele vem como _string_ e o _test_ quer ele como _number_ então já prepara essa mudança, um simples _if_, olhando a chave do _state_ e usando _Number( )_ no _value_ já basta.
```suggestion
handle(e, name) {
    this.setState({ [name]: e.target.value });
  }
```